### PR TITLE
[TS-696] The order for repeating groups verification in the report corresponds to the actual received message

### PR DESCRIPTION
**Pull request by yarchiksmith**

### CODE REVIEW

File: src/main/kotlin/com/exactpro/th2/check1/rule/AbstractCheckTask.kt [hunk 1]
```diff
@@ -386,7 +386,10 @@ abstract class AbstractCheckTask(
     protected fun RootMessageFilter.toCompareSettings(): ComparatorSettings =
         ComparatorSettings().also {
             it.metaContainer = VerificationUtil.toMetaContainer(this.messageFilter, false)
-            it.ignoredFields = this.comparisonSettings.ignoreFieldsList.toSet()
+            with(this.comparisonSettings) {
+                it.ignoredFields = this.ignoreFieldsList.toSet()
+                it.isKeepResultGroupOrder = true
+            }
```
Comment (Nikita-Smirnov-Exactpro) [inline src/main/kotlin/com/exactpro/th2/check1/rule/AbstractCheckTask.kt:392]:
```suggestion
            it.ignoredFields = this.comparisonSettings.ignoreFieldsList.toSet()
            it.isKeepResultGroupOrder = true
```

File: src/test/kotlin/com/exactpro/th2/check1/rule/check/TestCheckRuleTask.kt [hunk 2]
```diff
@@ -189,4 +203,61 @@ internal class TestCheckRuleTask : AbstractCheckTaskTest() {
...
+
+        checkTask(messageFilterForCheckOrder, eventID, streams).begin()
+
+        val eventBatches = awaitEventBatchRequest(1000L, 2)
+        val eventList = eventBatches.flatMap(EventBatch::getEventsList)
+        assertAll({
+            assertEquals(3, eventList.size)
+        }, {
+            val verificationEvents = eventList.filter { it.type == "Verification" }
+            assertEquals(1, verificationEvents.size)
```
Comment (Nikita-Smirnov-Exactpro) [inline src/test/kotlin/com/exactpro/th2/check1/rule/check/TestCheckRuleTask.kt:247]: How do we check the order of repeating groups if we submit only one filter?
checke1 sorts in order to increase the number of FAILED fields in the previous version.
Let's make a message with two group filters:
* both fields are failed
* the only one filed failed
and run it vs two messages which have (A, B) and (B, A) order of the repeating group.
The order of results will be equivalent the order in message

**Review by Nikita-Smirnov-Exactpro — CHANGES_REQUESTED** [review]

File: src/test/kotlin/com/exactpro/th2/check1/rule/check/TestCheckRuleTask.kt [hunk 3]
```diff
@@ -189,4 +205,95 @@ internal class TestCheckRuleTask : AbstractCheckTaskTest() {
...
+
+    companion object {
+        @JvmStatic
+        fun resultByFilter(): Stream<Arguments> {
+            return Stream.of(
+                Arguments.arguments(FAILED to mapOf("C" to "3".toValueFilter(), "D" to "4".toValueFilter())),
+                Arguments.arguments(SUCCESS to mapOf("A" to "1".toValueFilter(), "B" to "2".toValueFilter()))
+            )
+        }
+    }
```
Comment (OptimumOpium) [inline src/test/kotlin/com/exactpro/th2/check1/rule/check/TestCheckRuleTask.kt:298]: Where do you use this?

**Review by OptimumOpium — APPROVED** [review]

**Comment by yarchiksmith** [comment]
@Nikita-Smirnov-Exactpro Could you approve these changes?

**Pull request merged**